### PR TITLE
build(github): Add pytest matcher for Github Actions

### PR DESCRIPTION
**Pull request by billyvg**
This adds a matcher for pytest failures so that they are annotated in GitHub

### CODE REVIEW

File: tests/acceptance/test_api.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def setUp(self):
         self.path = "/api/applications/"
 
     def test_simple(self):
+        assert False
```
Comment (joshuarli) [inline tests/acceptance/test_api.py:35]: 🤔

Comment (billyvg) [reply tests/acceptance/test_api.py:35]: @joshuarli Just testing:

image

File: .github/pytest.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "pytest",
+      "pattern": [
+        {
+          "regexp": "^Traceback"
+        },
+        {
+          "regexp": "^\\s*File\\s\"(.*)\",\\sline\\s(\\d+),\\sin\\s(.*)$",
```
Comment (billyvg) [inline .github/pytest.json:10]:
```suggestion
          "regexp": "^\\s+File\\s\"(.*)\",\\sline\\s(\\d+),\\sin\\s(.*)$",
```

File: .github/pytest.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      "pattern": [
+        {
+          "regexp": "^Traceback"
+        },
+        {
+          "regexp": "^\\s+File\\s\"(.*)\",\\sline\\s(\\d+),\\sin\\s(.*)$",
+          "file": 1,
+          "message": 3,
+          "line": 2
+        }
```
Comment (billyvg) [inline .github/pytest.json:14]:
```suggestion
        },
        {
          "regexp": "^\\s+.*",
        }
```

**Comment by billyvg** [comment]
Unsure if this will be super useful since github problem matchers have a hard time with parsing multiline messages. I think ideally we'd show the stacktrace

**Comment by joshuarli** [comment]
> Unsure if this will be super useful since github problem matchers have a hard time with parsing multiline messages. I think ideally we'd show the stacktrace

Still, looks pretty nice. And a good PoC. Is there any additional context somewhere on why you're exploring this?

**Comment by billyvg** [comment]
@joshuarli nothing outside of trying to sell that GHA is worthwhile to move to.

**Review by joshuarli — APPROVED** [review]

**Pull request merged**